### PR TITLE
XAC controls for accessibility

### PR DESCRIPTION
**Pull request by benadida**
set up actions for keyboard keys that are meant to simulate XAC keys for testing.
* arrow keys as expected: ⬅️ ➡️ ⬆️ ⬇️
* `[` and `]` for B and A keys respectively.

This is implemented by doing some DOM navigation and clicking. There's an argument that this isn't the most modular React way to do this, but I'm not even sure what a correct React way to do this would be, and I suspect it would be a good bit gnarlier.

### CODE REVIEW

**Comment by benadida** [comment]
this now has all the XAC mappings. I've kept the keyboard mappings but have excluded that specific glue from test coverage, just to not overcomplicate tests with faking keyboard events that we're not going to keep long term.

File: src/pages/ContestPage.tsx [hunk 1]
```diff
@@ -56,26 +56,28 @@ const ContestPage = (props: Props) => {
...
           <LinkButton
+            id="previous"
             to={`/contests/${prevContest && prevContest.id}`}
             key="previous"
           >
             Previous
           </LinkButton>
         ) : (
-          <LinkButton key="backtostart" to="/start">
+          <LinkButton key="backtostart" to="/start" id="previous">
```
Comment (beausmith) [inline src/pages/ContestPage.tsx]: 🤔 Wondering if we should be concerned about these `id`'s being generic vs something more specific like `previous-contest`, `next-contest`, etc. No need to change now.

**Comment by beausmith** [comment]
Suggested changes: https://github.com/votingworks/bmd/commit/45e0d482d812fcb0a8c8c701b556db1b653b944e

Full branch: [xac-first-pass-plus](https://github.com/votingworks/bmd/compare/xac-first-pass-plus)

File: src/App.tsx [hunk 2]
```diff
@@ -169,19 +155,21 @@ class App extends React.Component<RouteComponentProps, State> {
...
+        <Gamepad onButtonDown={accessibility.gamepadButtonDown}>
+          <BallotContext.Provider
+            value={{
+              election,
+              resetBallot: this.resetBallot,
+              setBallotKey: this.setBallotKey,
             setUserSettings: this.setUserSettings,
-            updateVote: this.updateVote,
+              updateVote: this.updateVote,
             userSettings: this.state.userSettings,
```
[the remark this answers is not in the document]
Comment (benadida) [reply src/App.tsx]: weird. No idea. I'll see if it happens again.

File: src/accessibility.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+ *
+ * document.activeElement!
+ *
+ * Note the non-null typescript assertion.
+ *
+ * That is because, while typescript doesn't know there is always an activeElement, there actually always is.
+ * Adding a conditional would make it impossible to get to 100% test coverage, because we can never simulate that impossible situation.
+ */
+
+// because javascript has no true modulo
```
[the remark this answers is not in the document]
Comment (benadida) [reply src/accessibility.ts]: I think the comment on why I'm using `document.activeElement!` was important. But if you don't think it's needed, I can live without.

File: src/accessibility.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+function mod(n: number, m: number) {
+  return ((n % m) + m) % m
+}
+
+function get_focusable_elements() {
+  return Array.from(
+    document.querySelectorAll('button, input, [tabindex]:not([tabindex="-1"])')
```
[the remark this answers is not in the document]
Comment (benadida) [reply src/accessibility.ts]: the idea is to remain future-compatible with default browser behavior when pressing tab, if we add such elements.

File: src/accessibility.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+ * That is because, while typescript doesn't know there is always an activeElement, there actually always is.
+ * Adding a conditional would make it impossible to get to 100% test coverage, because we can never simulate that impossible situation.
+ */
+
+// because javascript has no true modulo
+function mod(n: number, m: number) {
+  return ((n % m) + m) % m
+}
+
+function get_focusable_elements() {
```
[the remark this answers is not in the document]
Comment (benadida) [reply src/accessibility.ts]: no good reason :)

File: src/App.tsx [hunk 6]
```diff
@@ -1,31 +1,17 @@
 import Mousetrap from 'mousetrap'
 import React from 'react'
+import Gamepad from 'react-gamepad'
```
[the remark this answers is not in the document]
Comment (benadida) [reply src/App.tsx]: 👍

File: src/pages/ContestPage.tsx (hunk 1 above)
Comment (benadida) [reply src/pages/ContestPage.tsx]: I think `next` makes sense if it is truly the next action in the overall app, which I think is the case here.

**Pull request merged**